pkg/resources/update_job.go

```diff
@@ -70,9 +73,22 @@ func (r *ResourceHelper) FetchMattermostUpdateJob(namespace string) (*batchv1.Jo
 	return job, err
 }
 
-func PrepareMattermostJobTemplate(name, namespace string, baseDeployment *appsv1.Deployment) *batchv1.Job {
+func PrepareMattermostJobTemplate(name, namespace string, baseDeployment *appsv1.Deployment, updateJobSpec *mmv1beta.UpdateJob) *batchv1.Job {
 	backoffLimit := int32(10)
 
+	podAnnotations := map[string]string{}
+	// Set user specified annotations
+	if updateJobSpec.ExtraAnnotations != nil {
```

Comment:
I think it would be good to do nil check here for `UpdateJob` then we can pass nil for ClusterInstallation.